JSQMessagesViewController/Controllers/JSQMessagesViewController.m

```diff
@@ -961,6 +988,33 @@ - (BOOL)jsq_isMenuVisible
 
 #pragma mark - Utilities
 
+/**
+ * Determines whether we should scroll to the end or not.
+ *
+ * Returns true as default.
+ */
+- (boolean_t)shouldScroll {
+    if (_delegate != nil) {
+        NSUInteger finalRow = MAX(0, [self.collectionView numberOfItemsInSection:0] - 1);
+        NSIndexPath *indexPath = [NSIndexPath indexPathForItem:finalRow inSection:0];
+        if ([_delegate respondsToSelector:@selector(shouldScrollToNewlyReceivedMessageAtIndexPath:)]) {
```

Comment:
since we're removing `@optional`, we can remove this check.

also, let's early return:

````objective-c
if (self.scrollingDelegate == nil) {
    return YES;
}
```